src/ngraph/runtime/reference/concat.hpp

```diff
@@ -39,6 +39,24 @@ namespace ngraph
 
                 for (size_t i = 0; i < args.size(); i++)
                 {
+                    // CoordinateTransform gets confused when the last input has a zero-size dim, so we will
+                    // just skip for zero-element tensors.
+                    bool any_zero = false;
+
+                    for (size_t len : in_shapes[i])
```

Comment:
I've seen this code somewhere before. You could make this a function `any_zeros(Shape)` if you feel like it. The code would read easier.

Reply:
I'll go ahead and do that, yeah. Thinking I'll test `shape_size(in_shapes[i]) == 0`.